apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/sites/SitesWebscript1.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.stereotype.Component;
+
+@WebScript(baseUri = RestV1Config.BaseUrl, families = RestV1Config.Family, defaultFormat = "json",
+        description = "Access operations on sites", value = "Sites")
+@Transaction(readOnly = false)
+@Component("eu.xenit.apix.rest.v1.SitesWebscript")
+public class SitesWebscript1 extends ApixV1Webscript {
+
+    private final static Logger logger = LoggerFactory.getLogger(SitesWebscript1.class);
+
+    @Autowired
+    INodeService nodeService;
+
+    @Autowired
+    IPermissionService permissionService;
+
+    @Autowired
+    IFileFolderService fileFolderService;
+
+    @Autowired
+    ISiteService siteService;
+
+    @Autowired
+    ServiceRegistry serviceRegistry;
+
+    @ApiOperation(value = "Retrieves information about the available sites of the current user",
+            notes = "Returns a list of sites. For each site the node reference, short name, title, description,\n"
+                    + "site visibility and list of site components (document libray, links, data lists, wiki,\n"
+                    + "discussions) are returned.\n"
+                    + "\n"
+                    + "There are no mandatory request parameters. However, there are optional ones:\n"
+                    + "Set 'retrieveMetadata' to true to return the aspects and properties of the sites.\n"
+                    + "Set 'retrievePath' to true to return the path of the sites.\n"
+                    + "Set 'retrievePermissions' to true to return the permissions of the sites.\n"
+                    + "Set 'retrieveChildAssocs' to true to return the child associations of the sites.\n"
+                    + "Set 'retrieveParentAssocs' to true to return the parent associations of the sites.\n"
+                    +"Set 'retrieveTargetAssocs' to true to return the peer associations of the sites.\n")
+    @Uri(value = "/sites/mySites", method = HttpMethod.GET)
+    @ApiResponses(@ApiResponse(code = 200, message = "Success", response = SiteInfo[].class))
+    public void getMySites(@RequestParam(required = false) Boolean retrieveMetadata,
+            @RequestParam(required = false) Boolean retrievePath,
+            @RequestParam(required = false) Boolean retrievePermissions,
+            @RequestParam(required = false) Boolean retrieveChildAssocs,
+            @RequestParam(required = false) Boolean retrieveParentAssocs,
+            @RequestParam(required = false) Boolean retrieveTargetAssocs,
+            WebScriptResponse response)
+            throws IOException {
+        logger.debug("retrieveMetadata: "+retrieveMetadata);
+        logger.debug("retrievePath: "+retrievePath);
+        logger.debug("retrievePermissions: "+retrievePermissions);
+        logger.debug("retrieveChildAssocs: "+retrieveChildAssocs);
+        logger.debug("retrieveParentAssocs: "+retrieveParentAssocs);
+        logger.debug("retrieveTargetAssocs: "+retrieveTargetAssocs);
+
+        if (retrieveMetadata == null){
```

Comment:
I would suggest to set the property defaultValue in @RequestParam (https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/bind/annotation/RequestParam.html) instead of writing null checks for each optional boolean request parameter.